x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryFolder.java

```diff
@@ -421,20 +422,20 @@ static EsQueryExec fold(AggregateExec a, EsQueryExec exec) {
 
             // track aliases defined in the SELECT and used inside GROUP BY
             // SELECT x AS a ... GROUP BY a
-            Map<Attribute, Expression> aliasMap = new LinkedHashMap<>();
             String id = null;
-            for (NamedExpression ne : a.aggregates()) {
-                if (ne instanceof Alias) {
-                    aliasMap.put(ne.toAttribute(), ((Alias) ne).child());
-                }
-            }
 
-            if (aliasMap.isEmpty() == false) {
-                Map<Attribute, Expression> newAliases = new LinkedHashMap<>(queryC.aliases());
-                newAliases.putAll(aliasMap);
-                queryC = queryC.withAliases(new AttributeMap<>(newAliases));
+            List<Alias> aliases = a.aggregates().stream()
+                .filter(ne -> ne instanceof Alias)
+                .map(ne -> (Alias)ne)
+                .collect(Collectors.toList());
```

Comment:
The initial code was using a `Map` of attributes, the change uses a `List` of `Alias`es which are then converted into a `Map`.
The use of streams doesn't seem to simplify things either - instead of one iteration there are too.
This needs reworking - just create an `AttributeMap` with the children and then combine that with the existing aliases.

Reply:
Ok, now that there is no chance of overwrite that makes more sense.